packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -143,30 +159,94 @@ contract ControllerHelper is FlashControllerHelper, IERC721Receiver {
      * @param _wPowerPerpAmountToBurn amount of WPowerPerp to burn
      * @param _wPowerPerpAmountToBuy amount of WPowerPerp to buy
      * @param _collateralToWithdraw amount of collateral to withdraw from vault
-     * @param _maxToPay max amount of collateral to pay for WPowerPerp token
+     * @param _collateralToBuyWith amount of collateral from vault to use to buy long
      */
     function flashswapWBurnBuyLong(
         uint256 _vaultId,
         uint256 _wPowerPerpAmountToBurn,
         uint256 _wPowerPerpAmountToBuy,
         uint256 _collateralToWithdraw,
-        uint256 _maxToPay
+        uint256 _collateralToBuyWith
     ) external payable {
-        require(_maxToPay <= _collateralToWithdraw.add(msg.value), "Not enough collateral");
+        require(_collateralToBuyWith <= _collateralToWithdraw.add(msg.value), "Not enough collateral");
```

Comment:
not sure why we made this change (do you know? did we discuss?), but think the require is not correct now at least if it means what I'm reading it to mean.

right now if the 5th param _collateralToBuyWith is the amount of vault collateral to use for buying oSQTH the require needs to be changed to be:

```
        require(_collateralToBuyWith <= _collateralToWithdraw, "Not enough collateral");

```

this is because you add msg.value to it later so it can use collateralToBuyWith + msg.value but still can't use more collateral than withdrawn

but honestly I think its more confusing this way - why did we change?

Reply:
I think this is changed in a rebase

Reply:
remove this change